### ISSUE TITLE
[Tablet Splitting] Unable to perform manual tablet splitting

### ISSUE THREAD

**Issue by Arjun-yb** [yugabyte/yugabyte-db#10347]
DB Version: **2.8.0.0-b2**

**Steps**: 
1. Created a database and table and inserted 1M rows.
```
demo=# CREATE TABLE t (k VARCHAR, v TEXT, PRIMARY KEY (k)) SPLIT INTO 1 TABLETS;
CREATE TABLE
demo=# INSERT INTO t (k, v)
demo-#     SELECT i::text, left(md5(random()::text), 4)
demo-#     FROM generate_series(1, 1000000) s(i);
INSERT 0 1000000
demo=# SELECT count(*) FROM t;
  count
---------
 1000000
(1 row)
```
2. Verified the table has one tablet
```
bin/yb-admin --master_addresses <master_ips> list_tablets ysql.demo t 
Tablet-UUID                      	Range                                                    	Leader-IP       	Leader-UUID 
77dc505657db42acb7d76bdb6270242b 	partition_key_start: "" partition_key_end: ""            	x.x.x.x:9100 	e4109e04f2f14d97b19d7a2aaaabcd79
```

3. Manually split the tablet
` bin/yb-admin --master_addresses <master_ips> split_tablet 77dc505657db42acb7d76bdb6270242b`
   And found below error
`    Error running split_tablet: Invalid argument (yb/master/catalog_manager.cc:2334): Unable to start split of tablet 77dc505657db42acb7d76bdb6270242b: Tablet split candidate 77dc505657db42acb7d76bdb6270242b is no longer a valid split candidate.  `

**Followed doc**: 
https://docs.yugabyte.com/latest/architecture/docdb-sharding/tablet-splitting/#manual-tablet-splitting

**Comment by robertsami** [comment]
looks like the error is triggered here https://github.com/yugabyte/yugabyte-db/blob/2.8/src/yb/master/catalog_manager.cc#L2331

this is because even for manual splits we still end up calling this method: https://github.com/yugabyte/yugabyte-db/blob/2.8/src/yb/master/catalog_manager.cc#L2282

looks like this is a bug introduced in fa415694e9209d827bd0ed0218b690f6d8cf4d90 and should be an easy fix. cc @SrivastavaAnubhav since you're revamping this code

**Comment by SrivastavaAnubhav** [comment]
Fixed by 55c034e. Backported to 2.8 in 6145a6c and to 2.12 in 9d3e848.
